transport/src/main/java/io/netty/channel/AbstractChannel.java

```diff
@@ -458,22 +468,22 @@ public final ChannelHandlerInvoker invoker() {
         }
 
         @Override
-        public final ChannelOutboundBuffer outboundBuffer() {
+        public ChannelOutboundBuffer outboundBuffer() {
             return outboundBuffer;
```

Comment:
In general I believe the third party should be able to override any of the methods. Although I require just one or two methods being made non-final, because of the above I decided to change most of them.  

Additionally, there is another problem with ChannelOutboundHandler that it's final and package private so basically third party framework does not have any way how to create it neither reimplement it.